webpack/assets/javascripts/react_app/components/HostsIndex/BulkActions/changeOwner/BulkChangeOwnerModal.js

```diff
@@ -0,0 +1,265 @@
+    setOwnerId(selection);
+    setOwnerSelectOpen(false);
+  };
+
+  const getOwnerLabel = id => {
+    if (id.endsWith('-Users')) {
+      const userId = id.replace('-Users', '');
+      return users?.results?.find(u => u.id.toString() === userId)?.login || id;
+    } else if (id.endsWith('-Usergroups')) {
+      const groupId = id.replace('-Usergroups', '');
+      return (
+        usergroups?.results?.find(ug => ug.id.toString() === groupId)?.name ||
+        id
+      );
+    }
+    return id;
+  };
+
+  const toggle = toggleRef => (
+    <MenuToggle
+      ref={toggleRef}
+      onClick={onToggleClick}
+      isExpanded={ownerSelectOpen}
+      style={{ width: '500px' }}
+    >
+      {ownerId ? getOwnerLabel(ownerId) : __('Select an owner')}
+    </MenuToggle>
+  );
+
+  const selectedTreeViewData = [
+    {
+      name: __('Selected hosts'),
+      id: 'selected-hosts-tree-view-title',
+      customBadgeContent: selectAllHostsMode ? 'All' : selectedCount,
+    },
+  ];
+
+  const handleModalClose = () => {
+    setOwnerId('');
+    closeModal();
+  };
+
+  const handleError = response => {
+    handleModalClose();
+    dispatch(
+      addToast(
+        failedHostsToastParams({
+          ...response.data.error,
+          key: BULK_CHANGE_OWNER_KEY,
+        })
+      )
+    );
+  };
+
+  const handleSuccess = response => {
+    dispatch(
+      addToast({
+        type: 'success',
+        message: response.data.message,
+      })
+    );
+    dispatch(
+      APIActions.get({
+        key: API_REQUEST_KEY,
+        url: foremanUrl(HOSTS_API_PATH),
+      })
+    );
+    handleModalClose();
+  };
+
+  const handleConfirm = () => {
+    const requestBody = {
+      included: {
+        search: fetchBulkParams(),
+      },
+      owner_id: ownerId,
+    };
+
+    dispatch(bulkChangeOwner(requestBody, handleSuccess, handleError));
+  };
+
+  const modalActions = [
+    <Button
+      key="add"
+      ouiaId="bulk-change-owner-modal-add-button"
+      variant="primary"
+      onClick={handleConfirm}
+      isDisabled={ownerId === ''}
+      isLoading={
+        userStatus === STATUS.PENDING || usergroupStatus === STATUS.PENDING
+      }
+    >
+      {__('Confirm')}
+    </Button>,
+    <Button
+      key="cancel"
+      ouiaId="bulk-change-owner-modal-cancel-button"
+      variant="link"
+      onClick={handleModalClose}
+    >
+      {__('Cancel')}
+    </Button>,
+  ];
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={handleModalClose}
+      onEscapePress={handleModalClose}
+      title={__('Change Owner')}
+      width="50%"
+      position="top"
+      actions={modalActions}
+      id="bulk-change-owner-modal"
+      key="bulk-change-owner-modal"
+      ouiaId="bulk-change-owner-modal"
+    >
+      <TextContent>
+        <Text ouiaId="bulk-change-owner-options">
+          {__(
+            'Changing owner will affect all your selected systems. Some systems may already been associated with your chosen owner.'
```

Comment:
Let's be consistent about what we call hosts

```suggestion
            'Changing owner will affect all your selected hosts. Some hosts may already been associated with your chosen owner.'
```